common/cpp/src/google_smart_card_common/numeric_conversions.h

```diff
@@ -135,6 +134,32 @@ inline bool CastIntegerToDouble(T value,
   return false;
 }
 
+// Assigns `*lhs` to `rhs`, after doing a compile-time assertion that the
+// destination type fits all possible values of the source.
+template <typename FirstType, typename SecondType>
+inline void AssignWithTypeSizeCheck(FirstType* lhs, const SecondType& rhs) {
+  GOOGLE_SMART_CARD_CHECK(lhs);
+  static_assert(std::is_integral<FirstType>::value,
+                "This assertion is only applicable to integral types");
+  static_assert(std::is_integral<SecondType>::value,
+                "This assertion is only applicable to integral types");
+  static_assert(sizeof(FirstType) > sizeof(SecondType) ||
+                    (sizeof(FirstType) == sizeof(SecondType) &&
+                     std::is_signed<FirstType>::value ==
+                         std::is_signed<SecondType>::value),
+                "Left-hand-side type is smaller than the right-hand-side type");
```

Comment:
nit: This error does not match the std::is_signed<FirstType>::value == std::is_signed<SecondType>::value check - I think we want a separate error message / assert for that

Reply:
Splitting static_assert conditions is somewhat hard, since all static_asserts are always executed (so you can't express "do this static_assert only if some other condition is true", at least not in C++11; "if constexpr" in C++17 could probably help us once we drop NaCl support).

The closest I can come up with is:
```
  static_assert(sizeof(FirstType) >= sizeof(SecondType), ...);
  static_assert(sizeof(FirstType) != sizeof(SecondType) ||
                         std::is_signed<FirstType>::value == std::is_signed<SecondType>::value,
                        ...);
```

Reply:
Published #478 - PTAL. Thanks.